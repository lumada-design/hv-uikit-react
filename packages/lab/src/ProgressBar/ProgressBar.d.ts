import * as React from "react";
import { StandardProps } from "@mui/material";

export type HvProgressBarClassKey =
  | "root"
  | "progress"
  | "progressContainer"
  | "progressBarContainer"
  | "progressBar"
  | "progressDone"
  | "progressError"
  | "progressBarLabel";

export interface HvProgressBarProps
  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, HvProgressBarClassKey> {
  /**
   * The value of the progress bar.
   */
  value?: number;
  /**
   * The status of the progress bar.
   *
   * inProgress is black, error is red and completed is green.
   *
   * When uncontrolled and unspecified it will default to "inProgress".
   */
  status?: string;
  /**
   * Aria Properties passed on to the progress bar.
   */
  labelProps?: object;
}

export default function HvProgressBar(props: HvProgressBarProps): JSX.Element | null;
