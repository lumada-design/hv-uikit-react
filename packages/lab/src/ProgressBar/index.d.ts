import * as React from "react";
import { StandardProps } from "@material-ui/core";

export type HvProgressClassKey =
  | "root"
  | "progress"
  | "progressContainer"
  | "progressBarContainer"
  | "progressBar"
  | "progressDone"
  | "progressError"
  | "progressBarLabel"
  | "progressAriaAlert";

export interface HvProgressBarProps
  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, HvProgressClassKey> {
  /**
   * The value of the progress bar.
   */
  value?: number;
  /**
   * Indicates current status of the progress bar represented by the color.
   */
  status?: boolean;
  /**
   * If `true` the progress bar will not show the percentage label.
   */
  undeterminate?: boolean;
  /**
   * Aria Properties passed on to the progress bar.
   */
  labelProps?: object;
}
