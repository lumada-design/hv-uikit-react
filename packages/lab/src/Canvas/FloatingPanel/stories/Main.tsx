import { css } from "@emotion/css";
import {
  HvCanvasFloatingPanel,
  HvCanvasFloatingPanelProps,
} from "@hitachivantara/uikit-react-lab";

const tabs = [
  {
    id: 0,
    title: "Tab 1",
  },
  {
    id: 1,
    title: "Tab 2",
  },
];

const classes = { root: css({ position: "relative" }) }; // for Storybook purposes

export const MainStory = (props: HvCanvasFloatingPanelProps) => (
  <HvCanvasFloatingPanel {...props} className={classes.root} tabs={tabs}>
    Content
  </HvCanvasFloatingPanel>
);
