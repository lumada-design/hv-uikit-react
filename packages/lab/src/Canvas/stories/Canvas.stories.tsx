import { useEffect, useState } from "react";
import { css } from "@emotion/css";
import { Meta, StoryObj } from "@storybook/react";
import { Background, BackgroundVariant, MiniMap } from "reactflow";
import {
  HvButton,
  HvDropDownMenu,
  HvIconButton,
  HvInlineEditor,
  HvLoading,
  theme,
} from "@hitachivantara/uikit-react-core";
import {
  Backwards,
  Calendar,
  DataSource,
  Plane,
  Pod,
  Redo,
  Schema,
  Undo,
  User,
} from "@hitachivantara/uikit-react-icons";
import {
  HvCanvasPanel,
  HvCanvasTab,
  HvCanvasToolbar,
  HvFlow,
} from "@hitachivantara/uikit-react-lab";

import { ListView } from "./ListView/ListView";
import { eds, nds, nodeTypes } from "./nodes";
import { TreeView } from "./TreeView";

const meta: Meta = {
  title: "Lab/Canvas",
};
export default meta;

export const Main: StoryObj = {
  args: {
    title: "Semantic Model Name",
  },
  argTypes: {
    classes: { control: { disable: true } },
  },
  render: () => {
    const [tabs, setTabs] = useState<HvCanvasTab[]>([]);
    const [selectedTab, setSelectedTab] = useState<string>("1");

    const classes = {
      separator: css({
        height: 30,
        width: 1,
        backgroundColor: theme.colors.atmo4,
        margin: `0 ${theme.space.xs}`,
      }),
    };
    const backButton = (
      <HvButton aria-label="Back" icon>
        <Backwards />
      </HvButton>
    );

    const title = (
      <HvInlineEditor value="Semantic Model Name" variant="title4" />
    );

    const Separator = () => <div className={classes.separator} />;

    const renderIcon = (Icon: React.ElementType) => () => <Icon />;

    useEffect(() => {
      setTabs([
        {
          id: "1",
          content: (
            <>
              <DataSource />
              Add Data
            </>
          ),
        },
        {
          id: "2",
          content: (
            <>
              <Schema />
              Model Structure
            </>
          ),
        },
      ]);
    }, []);

    return tabs && tabs.length > 0 ? (
      <div
        style={{
          width: "100%",
          height: "100vh",
          position: "relative",
        }}
      >
        <HvFlow
          style={{ width: "100%", height: "100vh" }}
          snapGrid={[1, 1]}
          snapToGrid
          nodes={nds}
          edges={eds}
          nodeTypes={nodeTypes}
        >
          <Background
            color={theme.palette.slate[950]}
            variant={BackgroundVariant.Dots}
            gap={16}
          />
          <MiniMap />
        </HvFlow>
        <HvCanvasPanel
          tabs={tabs}
          onTabChange={(tabId) => setSelectedTab(tabId)}
        >
          {selectedTab === "1" && <TreeView />}
          {selectedTab === "2" && <ListView />}
        </HvCanvasPanel>
        <HvCanvasToolbar backButton={backButton} title={title}>
          <HvIconButton title="Undo">
            <Undo />
          </HvIconButton>
          <HvIconButton title="Redo">
            <Redo />
          </HvIconButton>
          <Separator />
          <HvIconButton title="Add">
            <Pod />
          </HvIconButton>
          <Separator />
          <HvButton variant="primary">Save</HvButton>
          <HvButton variant="primaryGhost">Cancel</HvButton>
          <Separator />
          <HvDropDownMenu
            placement="right"
            onClick={(e, item) => console.log(item.label)}
            dataList={[
              { label: "Label 1", icon: renderIcon(User) },
              { label: "Label 2", icon: renderIcon(Calendar), disabled: true },
              { label: "Label 3", icon: renderIcon(Plane) },
            ]}
          />
        </HvCanvasToolbar>
      </div>
    ) : (
      <HvLoading />
    );
  },
};
