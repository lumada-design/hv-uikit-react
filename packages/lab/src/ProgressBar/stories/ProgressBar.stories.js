import React, { useEffect, useState } from "react";

import { HvButton, HvTypography } from "@hitachivantara/uikit-react-core";

import { HvProgressBar } from "../..";

export default {
  title: "Lab/ProgressBar",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvProgressBar } from '@hitachivantara/uikit-react-lab'",
  },
  component: HvProgressBar,
};
const ProgressBarSimulator = ({ inc, error, undeterminate, ariaLabel, ariaLive }) => {
  const [status, setStatus] = useState("inProgress");
  const [value, setValue] = useState(0);
  const [run, setRun] = useState(false);

  useEffect(() => {
    const interval = setInterval(() => {
      if (value === 100) {
        setRun(false);
        clearInterval(interval);
        setStatus("completed");
      } else if (error && value === error) {
        clearInterval(interval);
        setStatus("error");
      } else if (run) setValue(inc);
    }, 500);
    return () => clearInterval(interval);
  }, [inc, value, run, error]);

  const reset = () => {
    setValue(0);
    setStatus("inProgress");
  };

  const start = () => {
    setRun(true);
  };

  return (
    <div style={{ width: 400 }}>
      <HvProgressBar
        value={value}
        status={status}
        undeterminate={undeterminate}
        labelProps={{
          "aria-label": ariaLabel,
          "aria-busy": false,
          "aria-live": ariaLive,
        }}
      />
      <div style={{ marginTop: 10 }}>
        <HvButton onClick={start}>Start</HvButton>
        <HvButton style={{ marginLeft: 10 }} onClick={reset}>
          Reset
        </HvButton>
      </div>
    </div>
  );
};

export const Main = () => {
  return (
    <div
      style={{
        display: "flex:",
        flexDirection: "column",
      }}
    >
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: 400,
          margin: "auto",
          marginTop: 20,
          marginBottom: 40,
        }}
      >
        <HvTypography style={{ marginBottom: 15 }} variant="highlightText">
          Success Underterminate
        </HvTypography>
        <ProgressBarSimulator
          label={(v) => `${v}%`}
          inc={(v) => v + 5}
          undeterminate
          ariaLabel="Underterminate Progress Bar"
          ariaLive="polite"
        />
      </div>
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: 400,
          margin: "auto",
          marginTop: 20,
          marginBottom: 40,
        }}
      >
        <HvTypography variant="highlightText">Success</HvTypography>
        <ProgressBarSimulator
          label={(v) => `${v}%`}
          inc={(v) => v + 10}
          ariaLabel="Determinate Progress Bar"
          ariaLive="assertive"
        />
      </div>
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: 400,
          margin: "auto",
        }}
      >
        <HvTypography variant="highlightText">Error</HvTypography>
        <ProgressBarSimulator
          label={(v) => `${v}%`}
          inc={(v) => v + 5}
          error={30}
          ariaLabel="Determinate Progress Bar"
        />
      </div>
    </div>
  );
};

export const Determined = () => {
  return (
    <div
      style={{
        display: "flex:",
        flexDirection: "column",
      }}
    >
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: 400,
          margin: "auto",
          marginTop: 20,
          marginBottom: 40,
        }}
      >
        <HvTypography variant="highlightText">Success</HvTypography>
        <HvProgressBar
          value={100}
          status="completed"
          labelProps={{
            "aria-label": "Example Determined Progress Bar",
            // "aria-busy":run,
            "aria-live": "polite",
          }}
        />
      </div>
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: 400,
          margin: "auto",
          marginTop: 20,
          marginBottom: 40,
        }}
      >
        <HvTypography variant="highlightText">Loading</HvTypography>
        <HvProgressBar
          value={40}
          labelProps={{
            "aria-label": "Example Determined Loading Progress Bar",
            // "aria-busy":run,
            "aria-live": "polite",
          }}
        />
      </div>
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: 400,
          margin: "auto",
        }}
      >
        <HvTypography variant="highlightText">Error</HvTypography>
        <HvProgressBar
          value={30}
          status="error"
          labelProps={{
            "aria-label": "Example Determined Error Progress Bar",
            // "aria-busy":run,
            "aria-live": "polite",
          }}
        />
      </div>
    </div>
  );
};

Determined.parameters = {
  docs: {},
};

export const Undeterminate = () => {
  return (
    <div
      style={{
        display: "flex:",
        flexDirection: "column",
      }}
    >
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: 400,
          margin: "auto",
          marginTop: 20,
          marginBottom: 40,
        }}
      >
        <HvTypography style={{ marginBottom: 14 }} variant="highlightText">
          Success
        </HvTypography>
        <HvProgressBar
          status="completed"
          value={100}
          undeterminate
          labelProps={{
            "aria-label": "Example Undetermined Completed Progress Bar",
            // "aria-busy":run,
            "aria-live": "polite",
          }}
        />
      </div>
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: 400,
          margin: "auto",
          marginTop: 20,
          marginBottom: 40,
        }}
      >
        <HvTypography style={{ marginBottom: 14 }} variant="highlightText">
          Loading
        </HvTypography>
        <HvProgressBar
          value={40}
          undeterminate
          labelProps={{
            "aria-label": "Example Undetermined Loading Progress Bar",
            // "aria-busy":run,
            "aria-live": "polite",
          }}
        />
      </div>
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: 400,
          margin: "auto",
        }}
      >
        <HvTypography style={{ marginBottom: 14 }} variant="highlightText">
          Error
        </HvTypography>
        <HvProgressBar
          value={30}
          undeterminate
          status="error"
          labelProps={{
            "aria-label": "Example Undetermined Error Progress Bar",
            // "aria-busy":run,
            "aria-live": "polite",
          }}
        />
      </div>
    </div>
  );
};

Undeterminate.parameters = {
  docs: {},
};
