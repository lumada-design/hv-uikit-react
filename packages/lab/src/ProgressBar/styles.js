const styles = (theme) => ({
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
  },

  progress: {
    width: "100%",
  },

  progressContainer: {
    width: "100%",
  },

  progressBarContainer: {
    display: "flex",
    width: "100%",
    height: 4,
    backgroundColor: theme.hv.palette.atmosphere.atmo4,
  },

  progressBar: {
    backgroundColor: "#000",
    height: 4,
  },

  progressDone: {
    backgroundColor: theme.hv.palette.semantic.sema1,
  },

  progressError: {
    backgroundColor: theme.hv.palette.semantic.sema4,
  },

  progressBarLabel: {
    backgroundColor: "transparent",
    display: "flex",
    justifyContent: "end",
  },

  progressBarLabelHidden: {
    width: "1px",
    height: "1px",
    overflow: "hidden",
  },
});

export default styles;
