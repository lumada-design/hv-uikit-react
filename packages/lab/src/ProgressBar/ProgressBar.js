import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";

import clamp from "lodash/clamp";
import { HvTypography } from "@hitachivantara/uikit-react-core";

import { withStyles } from "@material-ui/core";

import styles from "./styles";

/**
 * ProgressBar provides feedback about a process that is taking place in the application.
 */
const HvProgressBar = (props) => {
  const {
    className,
    classes,
    value = 0,
    status,
    undeterminate = false,
    labelProps,
    ...others
  } = props;

  const clampedValue = clamp(value, 0, 100);

  return (
    <div
      className={clsx(className, classes.root, classes.progress)}
      role="progressbar"
      status="inProgress"
      aria-valuenow={clampedValue} // it only makes sense to put this in ariaProps if the min and max values of the progress bar are intended to be changed by the user
      aria-valuemin="0"
      aria-valuemax="100"
      {...others}
    >
      <div className={classes.progressContainer}>
        <HvTypography
          variant="vizText"
          style={{ width: `${clampedValue}%` }}
          className={clsx(
            classes.progressBarLabel,
            status === "completed" && classes.progressDone,
            undeterminate && classes.progressBarLabelHidden
          )}
          {...labelProps}
        >
          {`${clampedValue}%`}
        </HvTypography>
        <div className={classes.progressBarContainer}>
          <div
            style={{ width: `${clampedValue}%` }}
            className={clsx(
              classes.progressBar,
              status === "completed" && classes.progressDone,
              status === "error" && classes.progressError
            )}
          />
        </div>
      </div>
    </div>
  );
};

HvProgressBar.propTypes = {
  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the component root class.
     */
    root: PropTypes.string,

    /**
     * Style applied to the overall container when in progress mode.
     */
    progress: PropTypes.string,
    /**
     * Style applied to the specifc container when in progress mode.
     */
    progressContainer: PropTypes.string,
    /**
     * Style applied to progress bar container.
     */
    progressBarContainer: PropTypes.string,
    /**
     * Style applied to the progress bar.
     */
    progressBar: PropTypes.string,
    /**
     * Style applied to the progress bar when the loading is done.
     */
    progressDone: PropTypes.string,
    /**
     * Style applied to the progress bar when an error occurs.
     */
    progressError: PropTypes.string,
    /**
     * Style applied to the progress bar label.
     */
    progressBarLabel: PropTypes.string,
    /**
     * Style applied to the progress bar label.
     */
    progressBarLabelHidden: PropTypes.string,
  }).isRequired,

  /**
   * The value of the progress bar.
   */
  value: PropTypes.number,
  /**
   * The status of the progress bar.
   *
   * inProgress is black, error is red and completed is green.
   *
   * When uncontrolled and unspecified it will default to "inProgress".
   */
  status: PropTypes.oneOf(["inProgress", "completed", "error"]),
  /**
   * If `true` the progress bar will not show the percentage label.
   */
  undeterminate: PropTypes.bool,
  /**
   * Aria Properties passed on to the progress bar.
   */
  labelProps: PropTypes.instanceOf(Object),
};

export default withStyles(styles, { name: "HvProgressBar" })(HvProgressBar);
