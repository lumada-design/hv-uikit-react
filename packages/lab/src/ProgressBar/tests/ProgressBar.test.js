/* eslint-env jest */

import userEvent from "@testing-library/user-event";
import React, { useEffect, useState } from "react";

import { render, screen } from "testing-utils";
import { HvButton } from "@hitachivantara/uikit-react-core";

import { HvProgressBar } from "../..";

import { Main } from "../stories/ProgressBar.stories";

// eslint-disable-next-line react/prop-types
const ProgressBarSimulator = ({ inc, error, undeterminate, ariaLabel, ariaLive }) => {
  const [status, setStatus] = useState("inProgress");
  const [value, setValue] = useState(0);
  const [run, setRun] = useState(false);

  useEffect(() => {
    const interval = setInterval(() => {
      if (value === 100) {
        setRun(false);
        clearInterval(interval);
        setStatus("completed");
      } else if (error && value === error) {
        clearInterval(interval);
        setStatus("error");
      } else if (run) setValue(inc);
    }, 5);
    return () => clearInterval(interval);
  }, [inc, value, run, error]);

  const reset = () => {
    setValue(0);
    setStatus("inProgress");
  };

  const start = () => {
    setRun(true);
  };

  return (
    <div style={{ width: 400 }}>
      <HvProgressBar
        value={value}
        status={status}
        undeterminate={undeterminate}
        labelProps={{
          "aria-label": ariaLabel,
          "aria-busy": false,
          "aria-live": ariaLive,
        }}
      />
      <div style={{ marginTop: 10 }}>
        <HvButton onClick={start}>Start</HvButton>
        <HvButton style={{ marginLeft: 10 }} onClick={reset}>
          Reset
        </HvButton>
      </div>
    </div>
  );
};

describe("ProgressBar", () => {
  jest.setTimeout(30000);
  it("General", async () => {
    const { container } = render(<Main />);

    expect(container).toMatchSnapshot();
  });

  describe("Static Determinate Value Tests", () => {
    it("completed progress bar", async () => {
      const { container, findByRole, findByText, getByLabelText } = render(
        <HvProgressBar
          value={100}
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );

      const progressBar = await findByRole("progressbar");
      await findByText("100%");
      expect(progressBar).toHaveAttribute("aria-valuenow", "100");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).not.toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("zero progress bar", async () => {
      const { container, findByRole, findByText, getByLabelText } = render(
        <HvProgressBar
          value={0}
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );
      const progressBar = await findByRole("progressbar");
      await findByText("0%");
      expect(progressBar).toHaveAttribute("aria-valuenow", "0");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).not.toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("middle progress bar", async () => {
      const { container, findByRole, findByText, getByLabelText } = render(
        <HvProgressBar
          value={50}
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );
      const progressBar = await findByRole("progressbar");
      await findByText("50%");
      expect(progressBar).toHaveAttribute("aria-valuenow", "50");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).not.toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("out of bounds progress bar (up)", async () => {
      const { container, findByRole, findByText, getByLabelText } = render(
        <HvProgressBar
          value={200}
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );
      const progressBar = await findByRole("progressbar");
      await findByText("100%");
      expect(progressBar).toHaveAttribute("aria-valuenow", "100");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).not.toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("out of bounds progress bar (down)", async () => {
      const { container, findByRole, findByText, getByLabelText } = render(
        <HvProgressBar
          value={-200}
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );
      const progressBar = await findByRole("progressbar");
      await findByText("0%");
      expect(progressBar).toHaveAttribute("aria-valuenow", "0");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).not.toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("error", async () => {
      const { container, findByRole, findByText, getByLabelText } = render(
        <HvProgressBar
          value={-200}
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );
      const progressBar = await findByRole("progressbar");
      await findByText("0%");
      expect(progressBar).toHaveAttribute("aria-valuenow", "0");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).not.toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });
  });

  describe("Static Undeterminate Value Tests", () => {
    it("completed progress bar", async () => {
      const { container, findByRole, getByLabelText } = render(
        <HvProgressBar
          value={100}
          undeterminate
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );

      const progressBar = await findByRole("progressbar");
      expect(progressBar).toHaveAttribute("aria-valuenow", "100");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("zero progress bar", async () => {
      const { container, findByRole, getByLabelText } = render(
        <HvProgressBar
          value={0}
          undeterminate
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );

      const progressBar = await findByRole("progressbar");
      expect(progressBar).toHaveAttribute("aria-valuenow", "0");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("middle progress bar", async () => {
      const { container, findByRole, getByLabelText } = render(
        <HvProgressBar
          value={50}
          undeterminate
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );
      const progressBar = await findByRole("progressbar");
      expect(progressBar).toHaveAttribute("aria-valuenow", "50");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("out of bounds progress bar (up)", async () => {
      const { container, findByRole, getByLabelText } = render(
        <HvProgressBar
          value={200}
          undeterminate
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );
      const progressBar = await findByRole("progressbar");
      expect(progressBar).toHaveAttribute("aria-valuenow", "100");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("out of bounds progress bar (down)", async () => {
      const { container, findByRole, getByLabelText } = render(
        <HvProgressBar
          value={-200}
          undeterminate
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );
      const progressBar = await findByRole("progressbar");
      expect(progressBar).toHaveAttribute("aria-valuenow", "0");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });

    it("error", async () => {
      const { container, findByRole, getByLabelText } = render(
        <HvProgressBar
          value={-200}
          undeterminate
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );

      const progressBar = await findByRole("progressbar");
      expect(progressBar).toHaveAttribute("aria-valuenow", "0");

      const bar = getByLabelText("My Aria Label");
      expect(bar).toBeInTheDocument();
      expect(bar).toHaveClass(
        "HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden"
      );

      expect(container).toBeInTheDocument();
    });
  });

  describe("Aria Label", () => {
    it("check aria label", async () => {
      const { container, findByRole, getByLabelText } = render(
        <HvProgressBar
          value={100}
          labelProps={{
            "aria-label": "My Aria Label",
            "aria-busy": false,
            "aria-live": "polite",
          }}
        />
      );
      await findByRole("progressbar");

      const bar = getByLabelText("My Aria Label");

      expect(bar).toBeInTheDocument();

      expect(container).toBeInTheDocument();
    });
  });

  describe("Dynamic Determinate Value Tests", () => {
    it("completed progress bar", async () => {
      const { container, findByRole, findByText } = render(
        <ProgressBarSimulator label={(v) => `${v}%`} inc={(v) => v + 75} arialabel="Progress Bar" />
      );
      const progressBar = await findByRole("progressbar");
      const button = screen.getByText("Start");
      await findByText("0%");
      expect(progressBar).toHaveAttribute("aria-valuenow", "0");

      userEvent.click(button);
      await findByText("100%");
      expect(progressBar).toHaveAttribute("aria-valuenow", "100");
      expect(container).toBeInTheDocument();
    });

    it("error progress bar", async () => {
      const { container, findByRole, findByText } = render(
        <ProgressBarSimulator
          label={(v) => `${v}%`}
          inc={(v) => v + 15}
          arialabel="Progress Bar"
          error={30}
        />
      );
      const progressBar = await findByRole("progressbar");
      const button = screen.getByText("Start");
      await findByText("0%");
      userEvent.click(button);
      await findByText("30%");
      expect(progressBar).toHaveAttribute("aria-valuenow", "30");
      expect(container).toBeInTheDocument();
    });
  });
});
