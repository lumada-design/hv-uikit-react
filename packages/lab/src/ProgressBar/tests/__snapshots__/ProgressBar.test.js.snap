// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar General 1`] = `
<div>
  <div
    style="display: flex:; flex-direction: column;"
  >
    <div
      style="display: flex; flex-direction: column; justify-content: space-around; align-items: center; width: 400px; margin: 20px auto 40px auto;"
    >
      <p
        class="HvTypography-root HvTypography-highlightText"
        style="margin-bottom: 15px;"
      >
        Success Underterminate
      </p>
      <div
        style="width: 400px;"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="HvProgressBar-root HvProgressBar-progress"
          role="progressbar"
          status="inProgress"
        >
          <div
            class="HvProgressBar-progressContainer"
          >
            <p
              aria-busy="false"
              aria-label="Underterminate Progress Bar"
              aria-live="polite"
              class="HvProgressBar-progressBarLabel HvProgressBar-progressBarLabelHidden HvTypography-root HvTypography-vizText"
              style="width: 0%;"
            >
              0%
            </p>
            <div
              class="HvProgressBar-progressBarContainer"
            >
              <div
                class="HvProgressBar-progressBar"
                style="width: 0%;"
              />
            </div>
          </div>
        </div>
        <div
          style="margin-top: 10px;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-contained HvButton-primarySVG MuiButton-containedPrimary HvButton-primary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Start
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-contained HvButton-primarySVG MuiButton-containedPrimary HvButton-primary"
            style="margin-left: 10px;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Reset
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      style="display: flex; flex-direction: column; justify-content: space-around; align-items: center; width: 400px; margin: 20px auto 40px auto;"
    >
      <p
        class="HvTypography-root HvTypography-highlightText"
      >
        Success
      </p>
      <div
        style="width: 400px;"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="HvProgressBar-root HvProgressBar-progress"
          role="progressbar"
          status="inProgress"
        >
          <div
            class="HvProgressBar-progressContainer"
          >
            <p
              aria-busy="false"
              aria-label="Determinate Progress Bar"
              aria-live="assertive"
              class="HvProgressBar-progressBarLabel HvTypography-root HvTypography-vizText"
              style="width: 0%;"
            >
              0%
            </p>
            <div
              class="HvProgressBar-progressBarContainer"
            >
              <div
                class="HvProgressBar-progressBar"
                style="width: 0%;"
              />
            </div>
          </div>
        </div>
        <div
          style="margin-top: 10px;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-contained HvButton-primarySVG MuiButton-containedPrimary HvButton-primary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Start
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-contained HvButton-primarySVG MuiButton-containedPrimary HvButton-primary"
            style="margin-left: 10px;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Reset
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      style="display: flex; flex-direction: column; justify-content: space-around; align-items: center; width: 400px; margin: auto;"
    >
      <p
        class="HvTypography-root HvTypography-highlightText"
      >
        Error
      </p>
      <div
        style="width: 400px;"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="HvProgressBar-root HvProgressBar-progress"
          role="progressbar"
          status="inProgress"
        >
          <div
            class="HvProgressBar-progressContainer"
          >
            <p
              aria-busy="false"
              aria-label="Determinate Progress Bar"
              class="HvProgressBar-progressBarLabel HvTypography-root HvTypography-vizText"
              style="width: 0%;"
            >
              0%
            </p>
            <div
              class="HvProgressBar-progressBarContainer"
            >
              <div
                class="HvProgressBar-progressBar"
                style="width: 0%;"
              />
            </div>
          </div>
        </div>
        <div
          style="margin-top: 10px;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-contained HvButton-primarySVG MuiButton-containedPrimary HvButton-primary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Start
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-contained HvButton-primarySVG MuiButton-containedPrimary HvButton-primary"
            style="margin-left: 10px;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Reset
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
